datafusion/core/tests/parquet/page_pruning.rs

```diff
@@ -204,3 +222,466 @@ async fn page_index_filter_multi_col() {
+async fn prune_f64_scalar_fun() {
+    // result of sql "SELECT * FROM t where abs(f-1) <= 0.000001" is not supported
+    test_prune(
+        Scenario::Float64,
+        "SELECT * FROM t where abs(f-1) <= 0.000001",
+        Some(0),
+        Some(0),
+        1,
+    )
+    .await;
+}
+
+#[tokio::test]
+#[ignore]
+async fn prune_f64_complex_expr() {
+    // result of sql "SELECT * FROM t where f+1 > 1.1"" is not supported
+    test_prune(
+        Scenario::Float64,
+        "SELECT * FROM t where f+1 > 1.1",
+        Some(0),
+        Some(0),
+        9,
+    )
+    .await;
+}
+
+#[tokio::test]
+#[ignore]
+async fn prune_f64_complex_expr_subtract() {
+    // result of sql "SELECT * FROM t where 1-f > 1" is not supported
+    test_prune(
+        Scenario::Float64,
+        "SELECT * FROM t where 1-f > 1",
+        Some(0),
+        Some(0),
+        9,
+    )
+    .await;
+}
+
+#[tokio::test]
+//                      null count  min                                       max
+// page-0                         0  -5                                        -1
+// page-1                         0  -4                                        0
+// page-2                         0  0                                         4
+// page-3                         0  5                                         9
+async fn prune_int32_eq_in_list() {
+    // result of sql "SELECT * FROM t where in (1)"
+    test_prune(
+        Scenario::Int32,
+        "SELECT * FROM t where i in (1)",
+        Some(0),
+        Some(15),
+        1,
+    )
+    .await;
+}
+
+#[tokio::test]
+async fn prune_int32_eq_in_list_negated() {
+    // result of sql "SELECT * FROM t where not in (1)" prune nothing
+    test_prune(
+        Scenario::Int32,
+        "SELECT * FROM t where i not in (1)",
+        Some(0),
+        Some(0),
+        19,
+    )
+    .await;
+}
+
+#[tokio::test]
+async fn prune_decimal_lt() {
+    // The data type of decimal_col is decimal(9,2)
+    // There are three pages each 5 rows:
+    // [1.00, 6.00], [-5.00,6.00], [20.00,60.00]
+    test_prune(
+        Scenario::Decimal,
+        "SELECT * FROM t where decimal_col < 4",
+        Some(0),
+        Some(5),
+        6,
+    )
+    .await;
+    // compare with the casted decimal value
+    test_prune(
+        Scenario::Decimal,
+        "SELECT * FROM t where decimal_col < cast(4.55 as decimal(20,2))",
+        Some(0),
+        Some(5),
+        8,
+    )
+    .await;
+
+    // The data type of decimal_col is decimal(38,2)
+    test_prune(
+        Scenario::DecimalLargePrecision,
+        "SELECT * FROM t where decimal_col < 4",
+        Some(0),
+        Some(5),
+        6,
+    )
+    .await;
+    // compare with the casted decimal value
+    test_prune(
+        Scenario::DecimalLargePrecision,
+        "SELECT * FROM t where decimal_col < cast(4.55 as decimal(20,2))",
+        Some(0),
+        Some(5),
+        8,
+    )
+    .await;
+}
+
+#[tokio::test]
+async fn prune_decimal_eq() {
```

Comment:
it might be worth another test that prunes something other than 5 rows -- maybe `where decimal_col = 30.00` and prunes out the other pages? All of the tests here seem to prune out only the third page 20.00 -> 60.00